UI/BuildDesignatorWnd.cpp

```diff
@@ -241,25 +244,109 @@ namespace {
             }
         }
         TemporaryPtr<const UniverseObject> source = GetSourceObjectForEmpire(empire_id);
-        return ConditionDescription(location_conditions, GetUniverseObject(candidate_object_id), source);
+
+        if (only_failed_conditions)
+            return ConditionFailedDescription(location_conditions, GetUniverseObject(candidate_object_id), source);
+        else
+            return ConditionDescription(location_conditions, GetUniverseObject(candidate_object_id), source);
     }
 
     boost::shared_ptr<GG::BrowseInfoWnd> ProductionItemRowBrowseWnd(const ProductionQueue::ProductionItem& item,
                                                                     int candidate_object_id, int empire_id)
     {
+        std::string title;
+        std::string main_text;
+        float total_cost;
+        int production_time;
+
+        // production item is a building
         if (item.build_type == BT_BUILDING) {
+            const BuildingType* building_type = GetBuildingType(item.name);
+            if (!building_type) {
+                boost::shared_ptr<GG::BrowseInfoWnd> browse_wnd;
+                return browse_wnd;
+            }
+
+            // create title, description, production time and cost
+            title = UserString(item.name);
+            main_text = UserString(building_type->Description());
+            total_cost = building_type->ProductionCost(empire_id, candidate_object_id);
+            production_time = building_type->ProductionTime(empire_id, candidate_object_id);
+
+            main_text += "\n\n" + UserString("PRODUCTION_WND_TOOLTIP_PROD_COST") + ": " + DoubleToString(total_cost, 3, false);
+            main_text += "\n" + UserString("PRODUCTION_WND_TOOLTIP_PROD_TIME") + ": " + boost::lexical_cast<std::string>(production_time);
+
+            // show build conditions
+            std::string EnqueueAndLocationConditionFailedText = EnqueueAndLocationConditionDescription(item.name, candidate_object_id, empire_id, true);
```

Comment:
variable names should be lower_case_undescore_separated not UnseparatedCamelCase.